Content.Server/Shuttles/Systems/ShuttleSystem.cs

```diff
@@ -115,40 +116,44 @@ private void OnShuttleStartup(EntityUid uid, ShuttleComponent component, Compone
 
             if (component.Enabled)
             {
-                Enable(physicsComponent);
+                Enable(uid, physicsComponent);
             }
         }
 
-        public void Toggle(ShuttleComponent component)
+        public void Toggle(EntityUid uid, ShuttleComponent component)
         {
             if (!EntityManager.TryGetComponent(component.Owner, out PhysicsComponent? physicsComponent)) return;
 
             component.Enabled = !component.Enabled;
 
             if (component.Enabled)
             {
-                Enable(physicsComponent);
+                Enable(uid, physicsComponent);
             }
             else
             {
-                Disable(physicsComponent);
+                Disable(uid, physicsComponent);
             }
         }
 
-        private void Enable(PhysicsComponent component)
+        private void Enable(EntityUid uid, PhysicsComponent component)
         {
-            component.BodyType = BodyType.Dynamic;
-            component.BodyStatus = BodyStatus.InAir;
-            component.FixedRotation = false;
-            component.LinearDamping = ShuttleLinearDamping;
-            component.AngularDamping = ShuttleAngularDamping;
+            FixturesComponent? manager = null;
+
+            _physics.SetBodyType(uid, BodyType.Dynamic, manager: manager, body: component);
+            _physics.SetBodyStatus(component, BodyStatus.InAir);
+            _physics.SetFixedRotation(uid, false, manager: manager, body: component);
+            _physics.SetLinearDamping(component, ShuttleLinearDamping);
+            _physics.SetAngularDamping(component, ShuttleAngularDamping);
         }
 
-        private void Disable(PhysicsComponent component)
+        private void Disable(EntityUid uid, PhysicsComponent component)
         {
-            component.BodyType = BodyType.Static;
-            component.BodyStatus = BodyStatus.OnGround;
-            component.FixedRotation = true;
+            FixturesComponent? manager = null;
+
+            _physics.SetBodyType(uid, BodyType.Static, manager: manager, body: component);
+            _physics.SetBodyStatus(component, BodyStatus.OnGround);
+            _physics.SetFixedRotation(uid, false, manager: manager, body: component);
```

Comment:
old behavior was setting this to true, is this correct?

Reply:
oop